assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isRecord_Test.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2023 the original author or authors.
+ */
+package org.assertj.core.api.classes;
+
+import org.assertj.core.api.ClassAssert;
+import org.assertj.core.api.ClassAssertBaseTest;
+
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for <code>{@link org.assertj.core.api.ClassAssert#isRecord()}</code>.
+ *
+ * @author Louis Morgan
+ */
+public class ClassAssert_isRecord_Test extends ClassAssertBaseTest {
```

Comment:
Once the implementation is moved to AbstractClassAssert, this test class should contain concrete test cases that can run on Java 8 like you've done in Classes_assertIs*Record_Test.